src/easyscience/Objects/variable/descriptor_array.py

```
@@ -0,0 +1,760 @@
+        
+        # First use __div__ to compute `self / other`
+        # but first converting to the units of other
+        inverse_result = self._rsmooth_operator(other, op.truediv, units_must_match=False)
+        return inverse_result
+    
+    def __pow__(self, other: Union[DescriptorNumber, numbers.Number]) -> DescriptorArray:
+        """
+        Perform element-wise exponentiation with another DescriptorNumber or number.
+
+        :param other: The object to use as a denominator. Must be a number or DescriptorNumber with
+                    no unit or variance.
+        :return: A new DescriptorArray representing the result of the addition.
+        """
+        if not isinstance(other, (numbers.Number, DescriptorNumber)):
+            return NotImplemented
+
+        if isinstance(other, numbers.Number):
+            exponent = other
+        elif type(other) is DescriptorNumber:
+            if other.unit != 'dimensionless':
+                raise UnitError('Exponents must be dimensionless')
+            if other.variance is not None:
+                raise ValueError('Exponents must not have variance')
+            exponent = other.value
+        else:
+            return NotImplemented
+        try:
+            new_value = self.full_value**exponent
+        except Exception as message:
+            raise message from None
+        if np.any(np.isnan(new_value.values)):
+            raise ValueError('The result of the exponentiation is not a number')
+        descriptor_number = DescriptorArray.from_scipp(name=self.name, full_value=new_value)
+        descriptor_number.name = descriptor_number.unique_name
+        return descriptor_number
+
+    def __rpow__(self, other: numbers.Number) -> numbers.Number:
+        """
+        Defers reverse pow with a descriptor array, `a ** array`.
+        Exponentiation with regards to an array does not make sense,
+        and is not implemented.
+        """
+        return NotImplemented
```

I would probably raise an error here rather than return `NotImplemented`